tests/test_deploy.py

```diff
@@ -0,0 +1,175 @@
+    process = await asyncio.create_subprocess_exec(
+        "make",
+        "up-prod",
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+        cwd=osparc_gateway_server_root_dir,
+    )
+    stdout, stderr = await process.communicate()
+    assert process.returncode == 0, f"Unexpected error while deploying stack:\n{stderr}"
+    print(f"{stdout}")
+    print("--> osparc-dask-gateway stack deployed.")
+    healtcheck_endpoint = f"{dask_gateway_entrypoint}/api/health"
+    async for attempt in AsyncRetrying(
+        reraise=True, wait=wait_fixed(1), stop=stop_after_delay(60)
+    ):
+        with attempt:
+            print(
+                f"--> Connecting to {healtcheck_endpoint}, "
+                f"attempt {attempt.retry_state.attempt_number}...",
+            )
+            response = await aiohttp_client.get(healtcheck_endpoint)
+            response.raise_for_status()
+        print(
+            f"--> Connection to gateway server succeeded."
+            f" [{json.dumps(attempt.retry_state.retry_object.statistics)}]",
+        )
+
+    yield
+    print("<-- Stopping osparc-dask-gateway stack...")
+    process = await asyncio.create_subprocess_exec(
+        "make",
+        "down",
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+        cwd=osparc_gateway_server_root_dir,
+    )
+    stdout, stderr = await process.communicate()
+    assert process.returncode == 0, f"Unexpected error while deploying stack:\n{stderr}"
+    print(f"{stdout}")
+    print("<-- osparc-dask-gateway stack stopped.")
+
+
+async def test_deployment(
+    dask_gateway_stack_deployed_services,
+    dask_gateway_entrypoint: str,
+    faker: Faker,
+    dask_gateway_password: str,
+):
+    gateway = dask_gateway.Gateway(
+        address=dask_gateway_entrypoint,
+        auth=dask_gateway.BasicAuth(faker.pystr(), dask_gateway_password),
+    )
+
+    cluster = (
+        gateway.new_cluster()
+    )  # when returning we do have a new cluster (e.g. a new scheduler)
+
+    _NUM_WORKERS = 2
+    cluster.scale(
+        _NUM_WORKERS
+    )  # when returning we are in the process of creating the workers
+
+    # now wait until we get the workers
+    workers = None
+    async for attempt in AsyncRetrying(
+        reraise=True, wait=wait_fixed(1), stop=stop_after_delay(60)
+    ):
+        with attempt:
+            print(
+                f"--> Waiting to have {_NUM_WORKERS} running,"
+                f" attempt {attempt.retry_state.attempt_number}...",
+            )
+            assert "workers" in cluster.scheduler_info
+            assert len(cluster.scheduler_info["workers"]) == _NUM_WORKERS
+            workers = deepcopy(cluster.scheduler_info["workers"])
+            print(
+                f"!-- {_NUM_WORKERS} are running,"
+                f" [{json.dumps(attempt.retry_state.retry_object.statistics)}]",
+            )
+
+    # now check all this is stable
+    _SECONDS_STABLE = 6
```

Comment:
i do not understand the notation here?  you check six times that the info on workers fit ... why SECONDS? what does stability has to do there as well?

Reply:
you are right... was not thinking straight. it needs to sleep a bit in between.